wallet-new/integration/Functions.hs

```diff
@@ -326,6 +336,32 @@ runAction wc ws GetTransaction  = do
     -- Modify wallet state accordingly.
     pure $ ws
         & actionsNum +~ 1
+runAction wc ws ImportPoorWallet = do
+    poorSecret <- pickRandomElement =<< liftIO unsafePoorSecrets
+    withSystemTempFile "importWalletKey.sk" $ \tempFp _ -> do
+        liftIO $ usingLoggerName "integration-test" $ dumpPoorSecret tempFp poorSecret
+        poorWallet <- respToRes $ importWalletKey wc tempFp
+
+        checkInvariant
+            (walBalance poorWallet == minBound)
+            (PoorWalletBalanceIsZero poorWallet)
+
+        -- Modify wallet state accordingly.
+        pure $ ws
+            & wallets <>~ [poorWallet]
+            & actionsNum +~ 1
+  where
+    unsafePoorSecrets = usingLoggerName "integration-test" $ withConfigurations def $ pure $ fromMaybe (error "GeneratedSecrets are unknown") genesisSecretsPoor
+    dumpPoorSecret fp poorSec = let hdwSk = poorSecretToEncKey poorSec in
+        dumpUserSecret fp $
+        foldl' (.) identity [ usKeys %~ (hdwSk :)
```

Comment:
I really don't understand this. Is this actually a function composition in disguise?

Reply:
I took those bits of the code from `tools/src/keygen/Dump.hs` (as they are not exported)

Basically this part is mimicing some small functionality of `cardano-keygen` executable.

Reply:
@ksaric Yes -- observe the expansion:

```haskell
foldl' (.) identity [f, g, h]
-- foldl' k z (x : xs) = foldl' k (k z x) xs
foldl' (.) (identity . f) [g, h]
foldl' (.) (identity . f . g) [h]
foldl' (.) (identity . f . g . h) []
-- foldl' k z [] = z
(identity . f . g . h)
```